src/core/message-reply.ts

```diff
@@ -23,20 +16,20 @@ export function messageReplyFromError(e: unknown, code: number): GenericMessageR
 export type UnionMessageReply = GenericMessageReply & {
   /**
    * Resulting message entries or events returned from the invocation of the corresponding message.
-   * e.g. the resulting messages from a RecordsQuery
-   * Mutually exclusive with `data`.
+   * e.g. the resulting messages from a RecordsQuery, or array of messageCid strings for EventsGet or EventsQuery
+   * Mutually exclusive with `record`.
    */
-  entries?: QueryResultEntry[];
+  entries?: QueryResultEntry[] | ProtocolsConfigureMessage[] | MessagesGetReplyEntry[] | string[];
 
   /**
-   * Data corresponding to the message received if applicable (e.g. RecordsRead).
+   * Record corresponding to the message received if applicable (e.g. RecordsRead).
    * Mutually exclusive with `entries` and `cursor`.
    */
-  data?: Readable;
+  record?: RecordsWriteReply;
```

Comment:
Design: shouldn't this be called `RecordsReadReply`?

Even so it would still be odd because `Reply` types should ideally be of the same level as all other `Reply` types, not be a type for a nested property.

Can we do something like directly:

```suggestion
  record?: RecordsWriteMessage & {
    /**
     * The initial write of the record if the returned RecordsWrite message itself is not the initial write.
     */
    initialWrite?: RecordsWriteMessage;
    data: Readable;
  }
```

Reply:
Yeah I agree it's weird... It's because `RecordsReadReply` has a nested `record: RecordsWriteReply` as a property 😅 

Maybe this is an opportunity to rename `RecordsWriteReply`? It almost feels like it's just a `Record`, but I'm not sure if that's too general.

I'm ok with just making it explicit within the `UnionMessageReply` for now though.

Thoughts?

Reply:
I am also okay with just declaring the structure (like shown in the above) if that's what you mean?